platform/default/include/mbgl/gfx/headless_backend.hpp

```diff
@@ -0,0 +1,24 @@
+#pragma once
+
+#include <mbgl/gfx/renderable.hpp>
+#include <mbgl/gfx/renderer_backend.hpp>
+#include <mbgl/util/image.hpp>
+
+#include <memory>
+
+namespace mbgl {
+namespace gfx {
+
+class HeadlessBackend : public gfx::Renderable {
+public:
+    // Factory.
+    static std::unique_ptr<HeadlessBackend> makeBackend(Size = { 256, 256 }, gfx::ContextMode = gfx::ContextMode::Unique);
+    
+    HeadlessBackend(Size);
```

Comment:
protected?

Reply:
Done. Although it is abstract class with factory method, no need to promote super class protected.